libs/SalesforceNetwork/SalesforceNetwork/SalesforceNetwork/Classes/Network/Action/CSFAction.m

```diff
@@ -518,14 +519,23 @@ - (void)retry {
 }
 
 - (void)completeOperationWithError:(NSError *)error {
-    [self willChangeValueForKey:@"isExecuting"];
-    [self willChangeValueForKey:@"isFinished"];
-    _executing = NO;
-    _finished = YES;
-    self.error = error;
-    [self didChangeValueForKey:@"isExecuting"];
-    [self didChangeValueForKey:@"isFinished"];
+    if (_completeCalled) {
+        return;
+    }
+    else {
+        _completeCalled = YES;
+    }
+
+    if (self.isExecuting) {
```

Comment:
To avoid the "SFRestAPISalesforceAction 0x7be90060 went isFinished=YES without being started by the queue it is in" happening when running testBlockCancel" in the case where the action was cancelled before it was started